tools/mksyscall.c

```diff
@@ -450,6 +452,27 @@ static void generate_stub(int nfixed, int nparms)
 
   fprintf(stream, ")\n{\n");
 
+  /* Generate the result variable definition for non-void function */
+
+  if (strcmp(g_parm[RETTYPE_INDEX], "void") != 0)
+    {
+      fprintf(stream, "  uintptr_t result;\n\n");
```

Comment:
Let's always define result to simplify the code:
```
uintptr_t result = 0;
sched_note_syscall_enter(...);
...
sched_note_syscall_leave(..., result);
return result;
```

Reply:
I got it.